### PR TITLE
Feature/late registration penalty

### PR DESCRIPTION
**Pull request by FredrikMorstad**
## :sparkles: add support for public and registration opening date on frontend'
  - #### :warning: Dependant on td-org-uit-no/tdctl-api#54
Screenshot from 2023-01-30 11-52-10
Screenshot from 2023-01-30 11-52-51
### :nail_care:  Update styling and add possibility to add a register opening date(optional) on frontend
Screenshot from 2023-01-30 11-59-04
### :sparkles: Add system for detecting when refresh token has expired and set users to unauthenticated
 - Since there is no way for components to detect refresh token being expired, users can stay on a `privateRoute` without being authenticated

### CODE REVIEW

File: src/components/pages/profile/Profile.tsx [hunk 1]
```diff
@@ -15,15 +15,19 @@ const SettingsPage = () => {
...
+      const initalValues = {
+        name: response.realName ?? '',
+        email: response.email ?? '',
+        classof: response.classof ?? '',
+        phone: response.phone ?? '',
+      };
+      setInit(initalValues);
+      setActive(response.status);
+    } catch (error) {
+      console.log(error);
```
Comment (VetleWoie) [inline src/components/pages/profile/Profile.tsx:29]: Gracefully ignore error? :angel:

File: src/components/molecules/carousel/Carousel.tsx [hunk 2]
```diff
@@ -128,14 +119,15 @@ const Carousel: React.FC<CarouselProps> = ({
         <div
           className={styles.inner}
           style={{
-            transform: transformation,
+            // translate `translateLength` in x or y dir based on direction
```
Comment (VetleWoie) [inline src/components/molecules/carousel/Carousel.tsx:122]: Wohoo! Comments :clap:

File: src/routes/privateRoute.tsx [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+import React, { useContext, useEffect } from 'react';
 import { Redirect, Route, RouteProps } from 'react-router-dom';
 import { AuthenticateContext } from 'contexts/authProvider';
 
 interface PrivateRouteProps extends RouteProps {
   component: any;
 }
 
-//Protects protcted pages(login required)
+// Protects protected pages i.e login required
```
Comment (VetleWoie) [inline src/routes/privateRoute.tsx:9]: :clap:

File: src/utils/auth.ts [hunk 4]
```diff
@@ -23,10 +23,27 @@ const getRole = (): string => {
     const payload = jwt(accessToken) as TokenPayload;
     return payload['role'];
   } catch (err) {
-    throw err;
+    return 'unconfirmed';
   }
 };
 
+// gets the number of milliseconds until the refreshToken expires
```
Comment (VetleWoie) [inline src/utils/auth.ts:30]: :clap:

File: src/utils/auth.ts [hunk 5]
```diff
@@ -23,10 +23,27 @@ const getRole = (): string => {
...
+const findRefreshTokenExpInMs = (): number | undefined => {
+  const { refreshToken } = getTokens();
+  if (refreshToken === '') {
+    return undefined;
+  }
+  const payload = jwt(refreshToken) as TokenPayload;
+  // new date is based on milliseconds while jwt is based on second
```
Comment (VetleWoie) [inline src/utils/auth.ts:37]: :clap:

**Review by VetleWoie — APPROVED** [review]
Good job! :+1: Looks very good.
There was one error that was only logged to the console. Is this intended, or is it a minor slip-up?

I'll approve the request, and then you can merge it if it was intended.

File: src/components/molecules/carousel/Carousel.tsx (hunk 2 above)
Comment (FredrikMorstad) [reply src/components/molecules/carousel/Carousel.tsx:122]: ups, this should be removed :wink:

Comment (VetleWoie) [reply src/components/molecules/carousel/Carousel.tsx:122]: ... :cry:

File: src/components/pages/profile/Profile.tsx (hunk 1 above)
Comment (FredrikMorstad) [reply src/components/pages/profile/Profile.tsx:29]: yes, can be fixed if it turns out someone got to a protected page without being logged in

**Pull request merged**